students/csdotson/lesson05/mailroom_3.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+# Lesson 4 - Mailroom... part 2
+
+### Function Definitions ###
+def prompt(prompt, menu):
+    # Create menu based on arguments
+    while True:
+        try:
+            response = input(prompt)
+            if menu[response]() == "quit":
+                break
+        except KeyError:
+            print('Please enter a valid selection!')
+
+
+def send_thank_you():
+    # Create sub_prompt for creating thank you's and adding new donors/donations
+    prompt(thank_you_prompt, thank_you_menu)
+
+
+def add_donation():
+    # Append donation to existing donor, or add donor/donation to existing list
+    try:
+        donor_name = input("\nPlease enter a donor name: ")
+        new_donation = input(f"Please enter a donation amount for {donor_name}: ")
```

Comment:
I would suggest doing the float conversion here, so it's only happening once in the try block. You generally want to include only the code that might raise the exception in the try block if you can, and catch the exception as closely to that point as you can, so I'd move the dict look up and additions out since there's nothing in those operations that would raise a `ValueError`.